server/channels/app/web_broadcast_hooks.go

```diff
@@ -130,6 +133,40 @@ func (h *postedAckBroadcastHook) Process(msg *platform.HookedWebSocketEvent, web
 	return nil
 }
 
+func usePermalinkHook(message *model.WebSocketEvent, previewChannel *model.Channel, postJSON string) {
+	message.GetBroadcast().AddHook(broadcastPermalink, map[string]any{
+		"preview_channel": previewChannel,
```

Comment:
Noting that in a HA scenario we'd be serializing the full channel struct here, while in reality we only need a couple of fields. Hopefully not a big deal.

Reply:
Yes, this is only for permalink posts. So I think it should be fine. Otherwise we have to rework the `HasPermissionToReadChannel` method to work with specific fields.